plugins/org.wso2.developerstudio.eclipse.esb.project/src/org/wso2/developerstudio/eclipse/esb/project/ui/wizard/ConnectorFromFileSystemDialog.java

```diff
@@ -107,10 +112,22 @@ public void modifyText(ModifyEvent evt) {
         btnBrowse.addSelectionListener(new SelectionAdapter() {
 
             public void widgetSelected(SelectionEvent e) {
-                FileDialog fileDlg = new FileDialog(getShell());
+                FileDialog fileDlg = new FileDialog(getShell(), SWT.MULTI);
                 String fileName = fileDlg.open();
+
                 if (fileName != null) {
-                    txtCloudConnectorPath.setText(fileName);
+                    // Store the file names in an array
+                    String[] connectorPaths = fileDlg.getFileNames();
+                    String displayText = "";
+
+                    for (int i=0; i<connectorPaths.length;i++) {
+                        if (i!=0) {
+                            displayText += ", " + fileDlg.getFilterPath() + File.separator + connectorPaths[i];
```

Comment:
Shall we move fileDlg.getFilterPath() + File.separator to a variable before the for loop?